addons/website_mail_channel/controllers/main.py

```diff
@@ -175,8 +175,19 @@ def thread_discussion(self, group, message, mode='thread', date_begin=None, date
             base_domain = [('model', '=', 'mail.channel'), ('res_id', '=', group.id)]
         next_message = Message.search(base_domain + [('date', '<', message.date)], order="date DESC", limit=1) or None
         prev_message = Message.search(base_domain + [('date', '>', message.date)], order="date", limit=1) or None
+
+        # handle access to attachments for public / portal
+        if request.env.user.has_group('base.group_user'):
```

Comment:
`not _is_public()` would be matching `group_portal` (or any other group really) whereas `group_user` is just strictly checking for internal users.

However if we want to not use groups this condition is the definition of `not user.share` but I wish instead of being called share it was inverted and called `is_internal`.